### DISCUSSION TITLE
2 ESP Homkeys / Door / Garage

### DISCUSSION THREAD

**Discussion by fisch55** [rednblkx/HomeKey-ESP32#109] [General]
Hello everyone, I wanted to put my second ESP32 Homkey for the garage into operation today.
I already have one on my front door in conjunction with Homeassistant.

After I set up the second one, it always performs the same actions as the first one - i.e. opens the door. I would like to use it to open the garage via automation. Is that even possible or is only an ESP key possible here?

Thanks for your help...

**Comment by rednblkx** [comment]
Hi!

In order to add a second device, just modify the “MQTT client id” and all topics so it can communicate separately.

**Comment by fisch55** [comment]
Like this ?! 
<img width="530" alt="Bildschirmfoto 2024-08-15 um 22 50 38" src="https://github.com/user-attachments/assets/12c71965-eaad-4121-bf47-40c84442c9eb">

**Reply by fisch55** [reply]
Useing this as an automation in Homeassiant... If I scan the second device, it reacts like the first....opening the front door
<img width="1079" alt="Bildschirmfoto 2024-08-15 um 22 53 10" src="https://github.com/user-attachments/assets/f5b25b72-92b6-4e36-ae59-d79f164a18ba">

**Reply by rednblkx** [reply]
ah yes, if you use tags then it would be triggering the same automation since you would be using the same homekey. Hass doesn't care about the source since the ID of the tag is what matters. 

Instead you could use MQTT directly like this

![image](https://github.com/user-attachments/assets/6e074e7c-3a66-4e7b-b6ca-be3e02ef4cbc)

Just replace the topic with the one you defined above in "HK Auth Topic"

**Comment by fisch55** [comment]
Cool !!! THX a lot .-. It Works now like a charm!
